scripts/global.js

```diff
@@ -960,37 +961,24 @@ export async function updateEncryptionGUI(fEncrypted = null) {
  * @param {number} nFixedFee - An optional fixed satoshi fee
  * @returns {Promise<string|false>} - TXID on success, false or error on failure
  */
-export async function sweepAddress(arrUTXOs, sweepingMasterKey, nFixedFee = 0) {
-    const cTx = new bitjs.transaction();
-
-    // Load all UTXOs as inputs
-    let nTotal = 0;
-    for (const cUTXO of arrUTXOs) {
-        nTotal += cUTXO.sats;
-        cTx.addinput({
-            txid: cUTXO.id,
-            index: cUTXO.vout,
-            script: cUTXO.script,
-        });
-    }
-
-    // Use a given fixed fee, or use the network fee if unspecified
-    const nFee = nFixedFee || getNetwork().getFee(cTx.serialize().length);
-
-    // Use a new address from our wallet to sweep the UTXOs in to
-    const strAddress = (
-        await getNewAddress({ updateGUI: true, verify: false, nReceiving: 1 })
-    )[0];
-
-    // Sweep the full funds amount, minus the fee, leaving no change from any sweeped UTXOs
-    cTx.addoutput(strAddress, (nTotal - nFee) / COIN);
+export async function sweepAddress(arrUTXOs, sweepingMasterKey) {
+    const txBuilder = TransactionBuilder.create().addUTXOs(arrUTXOs);
+
+    const outputValue = txBuilder.valueIn - txBuilder.getFee();
+    const [address] = wallet.getNewAddress(1);
+    const tx = txBuilder
+        .addOutput({
+            address,
+            value: outputValue,
+        })
+        .build();
```

Comment:
The optional 'fixed fee' was removed from here, but it's still necessary if we want PIVCards to function identically, it was designed so that when a user redeems a promo, they redeem the precise amount designated by the creator (since a solid fee would be included in the value from the creator's client, deducted by the redeemer's client).